subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/JvmComponentDependencies.java

```diff
@@ -16,109 +16,67 @@
 
 package org.gradle.api.plugins.jvm;
 
-import org.gradle.api.Action;
 import org.gradle.api.Incubating;
 import org.gradle.api.Project;
 import org.gradle.api.artifacts.Dependency;
 import org.gradle.api.artifacts.ModuleDependency;
 import org.gradle.api.artifacts.ProjectDependency;
+import org.gradle.api.artifacts.dsl.DependencyAdder;
 
 /**
  * This DSL element is used to add dependencies to a component, like {@link JvmTestSuite}.
  *
  * <ul>
  *     <li><code>implementation</code> dependencies are used at compilation and runtime.</li>
  *     <li><code>compileOnly</code> dependencies are used only at compilation and are not available at runtime.</li>
- *     <li><code>runtimeOnly</code> dependencies are not available at  compilation and are used only at runtime.</li>
+ *     <li><code>runtimeOnly</code> dependencies are not available at compilation and are used only at runtime.</li>
+ *     <li><code>annotationProcessor</code> dependencies are used only at compilation, and are added to the annotation processor classpath</li>
  * </ul>
  *
- * @since 7.3
- *
  * @see org.gradle.api.artifacts.dsl.DependencyHandler For more information.
+ * @since 7.3
  */
 @Incubating
 public interface JvmComponentDependencies {
```

Comment:
Can we add `JvmComponentDependencies` and `DependencyAdder` XML files into `src/docs/dsl/` like we have for `org.gradle.api.artifacts.Configuration.xml`?  And then in `dsl.xml`, add these types to the "Core types" section for now.

Reply:
Done.